dgraph/cmd/mcp/mcp_server.go

```diff
@@ -102,22 +102,22 @@ func NewMCPServer(connectionString string, readOnly bool) (*server.MCPServer, er
 		s.AddTool(alterSchemaTool, func(ctx context.Context, request mcp.CallToolRequest) (*mcp.CallToolResult, error) {
 			schema, ok := request.GetArguments()["schema"].(string)
 			if !ok {
-				return nil, fmt.Errorf("schema must be present")
+				return mcp.NewToolResultError("Schema must be present"), nil
```

Comment:
Handling errors via mcp.NewToolResultError is the correct mode. If we return an `err`, the SSE server internals will return a 500 server error.